lib/xchain/provider/provider.go

```diff
@@ -150,14 +150,21 @@ func (p *Provider) stream(
 				ConfLevel: req.ConfLevel,
 			}
 
-			xBlock, exists, err := p.GetBlock(ctx, fetchReq)
-			if err != nil {
-				return nil, err
-			} else if !exists {
-				return nil, nil
+			var lastErr error
+			retryCount := 5
+			for i := 0; i < retryCount; i++ {
+				xBlock, exists, err := p.GetBlock(ctx, fetchReq)
+				if err != nil {
+					lastErr = err
+					time.Sleep(100 * time.Millisecond)
```

Comment:
we could also stick to `backoff := expbackoff.New(ctx, expbackoff.WithPeriodic(time.Milli*100)` then all our backoff logic follows the same pattern. Benefit is that sleep is aborted when context is cancelled.